src/components/LayerCollection/LayerPOLine.js

```diff
@@ -170,29 +170,34 @@ function LayerPOLine({
     [memoizedMutator.lineOrder, order, sendCallout],
   );
 
-  const submitPOLine = useCallback(({ saveAndOpen, ...line }) => {
+  const submitPOLine = useCallback(async ({ saveAndOpen, ...line }) => {
     setSavingValues(line);
     setIsLoading(true);
     const newLine = cloneDeep(line);
+    let savedLine;
 
-    return memoizedMutator.poLines
-      .POST(newLine)
-      .then(({ id: poLineId }) => Promise.all([poLineId, openOrder(saveAndOpen)]))
-      .then(([poLineId]) => {
-        sendCallout({
-          message: <FormattedMessage id="ui-orders.line.create.success" />,
-          type: 'success',
-        });
+    try {
+      savedLine = await memoizedMutator.poLines.POST(newLine);
 
-        history.push({
-          pathname: `/orders/view/${id}/po-line/view/${poLineId}`,
-          search,
-        });
-      })
-      .catch((e) => {
-        setIsLoading(false);
-        handleErrorResponse(e, line);
+      await openOrder(saveAndOpen);
+
+      sendCallout({
+        message: <FormattedMessage id="ui-orders.line.create.success" />,
+        type: 'success',
+      });
+
+      history.push({
+        pathname: `/orders/view/${id}/po-line/view/${savedLine.id}`,
+        search,
       });
+    } catch (e) {
+      if (saveAndOpen && savedLine) {
+        await memoizedMutator.poLines.DELETE(savedLine);
```

Comment:
What if this fails too?
(it's a catch block)

Reply:
nothing is happened, don't think it's good idea to show line is not remove as we couldn't open line